analyzers/src/SonarAnalyzer.CSharp/Rules/MutableFieldsShouldNotBe.cs

```diff
@@ -48,63 +45,44 @@ public abstract class MutableFieldsShouldNotBe : SonarDiagnosticAnalyzer
                 KnownType.System_Collections_Immutable_IImmutableList_T,
                 KnownType.System_Collections_Immutable_IImmutableSet_T,
                 KnownType.System_Collections_Immutable_IImmutableStack_T,
-                KnownType.System_Collections_Immutable_IImmutableQueue_T
-            );
+                KnownType.System_Collections_Immutable_IImmutableQueue_T);
 
         protected abstract ISet<SyntaxKind> InvalidModifiers { get; }
 
-        protected sealed override void Initialize(SonarAnalysisContext context)
-        {
-            context.RegisterSyntaxNodeActionInNonGenerated(CheckForIssue,
-                SyntaxKind.ClassDeclaration,
-                SyntaxKind.StructDeclaration);
-        }
+        protected sealed override void Initialize(SonarAnalysisContext context) =>
+            context.RegisterSyntaxNodeActionInNonGenerated(CheckForIssue, SyntaxKind.ClassDeclaration, SyntaxKind.StructDeclaration);
 
         private void CheckForIssue(SyntaxNodeAnalysisContext analysisContext)
         {
             var typeDeclaration = (TypeDeclarationSyntax)analysisContext.Node;
             var fieldDeclarations = typeDeclaration.Members.OfType<FieldDeclarationSyntax>();
 
-            var assignmentsImmutability = GetFieldAssignmentImmutability(typeDeclaration, fieldDeclarations,
-                analysisContext.SemanticModel);
+            var assignmentsImmutability = FieldAssignmentImmutability(typeDeclaration, fieldDeclarations, analysisContext.SemanticModel);
 
             foreach (var fieldDeclaration in fieldDeclarations)
             {
-                if (!HasAllInvalidModifiers(fieldDeclaration) ||
-                    fieldDeclaration.Declaration.Variables.Count == 0)
-                {
-                    return;
-                }
-
-                if (!(analysisContext.SemanticModel.GetDeclaredSymbol(fieldDeclaration.Declaration.Variables[0]) is IFieldSymbol fieldSymbol) ||
-                    fieldSymbol.Type == null ||
-                    fieldSymbol.GetEffectiveAccessibility() != Accessibility.Public ||
-                    IsImmutableOrValidMutableType(fieldSymbol.Type))
+                if (HasAllInvalidModifiers(fieldDeclaration)
+                    && fieldDeclaration.Declaration.Variables.Count > 0
+                    && analysisContext.SemanticModel.GetDeclaredSymbol(fieldDeclaration.Declaration.Variables[0]) is IFieldSymbol { Type: not null } fieldSymbol
```

Comment:
These two conditions are not covered, I could not come up even with broken syntax that would cover it.